client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js

```diff
@@ -0,0 +1,258 @@
+				.map( ( [ method, enabled ] ) => enabled && method )
+				.filter( Boolean ),
+		[ paymentMethodsState ]
+	);
+
+	const unCheckedPaymentMethods = Object.entries( paymentMethodsState )
+		.map( ( [ method, enabled ] ) => ! enabled && method )
+		.filter( Boolean );
+
+	const handleContinueClick = useCallback( () => {
+		// creating a separate callback, so that the main thread isn't blocked on click of the button
+		const callback = async () => {
+			updateEnabledPaymentMethodIds( [
+				// adding the newly selected payment methods and removing them from the `initialEnabledPaymentMethodIds` if unchecked
+				...new Set(
+					[
+						...initialEnabledPaymentMethodIds,
+						...checkedPaymentMethods,
+					].filter(
+						( method ) =>
+							! unCheckedPaymentMethods.includes( method )
+					)
+				),
+			] );
+
+			const isSuccess = await saveSettings();
+			if ( ! isSuccess ) {
+				// restoring the state, in case of soft route
+				updateEnabledPaymentMethodIds( initialEnabledPaymentMethodIds );
+				return;
+			}
+
+			setCompleted( true, 'setup-complete' );
+		};
+
+		callback();
+	}, [
+		unCheckedPaymentMethods,
+		checkedPaymentMethods,
+		updateEnabledPaymentMethodIds,
+		saveSettings,
+		setCompleted,
+		initialEnabledPaymentMethodIds,
+	] );
+
+	return (
+		<Button
+			isBusy={ isSaving }
+			disabled={ isSaving || 1 > checkedPaymentMethods.length }
+			onClick={ handleContinueClick }
+			isPrimary
+		>
+			{ __( 'Add payment methods', 'woocommerce-payments' ) }
+		</Button>
+	);
+};
+
+const AddPaymentMethodsTask = () => {
+	const availablePaymentMethods = useGetAvailablePaymentMethodIds();
+	const { isActive } = useContext( WizardTaskContext );
+
+	// I am using internal state in this component
+	// and committing the changes on `initialEnabledPaymentMethodIds` only when the "continue" button is clicked.
+	// Otherwise a user could navigate to another page via soft-routing and the settings would be in un-saved state,
+	// possibly causing errors.
+	const [
+		paymentMethodsState,
+		handlePaymentMethodChange,
+	] = usePaymentMethodsCheckboxState();
+	const selectedMethods = useMemo(
+		() =>
+			Object.entries( paymentMethodsState )
+				.map( ( [ method, enabled ] ) => enabled && method )
+				.filter( Boolean ),
+		[ paymentMethodsState ]
+	);
+
+	return (
+		<WizardTaskItem
+			className="add-payment-methods-task"
+			title={ __(
+				'Boost your sales with payment methods',
+				'woocommerce-payments'
+			) }
+			index={ 2 }
+		>
+			<CollapsibleBody>
+				<p className="wcpay-wizard-task__description-element is-muted-color">
+					{ interpolateComponents( {
+						mixedString: __(
+							'For best results, we recommend adding all available payment methods. ' +
+								"We'll only show your customer the most relevant payment methods " +
+								'based on their location. {{learnMoreLink /}}.',
+							'woocommerce-payments'
+						),
+						components: {
+							// TODO
+							learnMoreLink: (
+								<a href="?TODO">
```

Comment:
We don't have this yet. Here's the plan if you're curious: https://github.com/Automattic/woocommerce-payments/issues/2170#issuecomment-880008439